# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCDTMFSender.cpp
# Commit: 0991ebfeb365
# Full Hash: 0991ebfeb3657ca4e5c379c5211f6f4d1e29968d
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Make RTCDTMFSender signal DTMF tones through a MediaEvent. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112859
# ==============================================================================

diff -r bd3db9010131 -r 0991ebfeb365 dom/media/webrtc/jsapi/RTCDTMFSender.cpp
--- a/dom/media/webrtc/jsapi/RTCDTMFSender.cpp	Sun Apr 18 21:47:29 2021 +0200
+++ b/dom/media/webrtc/jsapi/RTCDTMFSender.cpp	Sun Apr 18 21:06:49 2021 +0200
@@ -29,11 +29,8 @@
 LazyLogModule gDtmfLog("RTCDTMFSender");
 
 RTCDTMFSender::RTCDTMFSender(nsPIDOMWindowInner* aWindow,
-                             TransceiverImpl* aTransceiver,
-                             AudioSessionConduit* aConduit)
-    : DOMEventTargetHelper(aWindow),
-      mTransceiver(aTransceiver),
-      mConduit(aConduit) {}
+                             TransceiverImpl* aTransceiver)
+    : DOMEventTargetHelper(aWindow), mTransceiver(aTransceiver) {}
 
 JSObject* RTCDTMFSender::WrapObject(JSContext* aCx,
                                     JS::Handle<JSObject*> aGivenProto) {
@@ -67,6 +64,13 @@
   return !recognized[c];
 }
 
+void RTCDTMFSender::SetPayloadType(int32_t aPayloadType,
+                                   int32_t aPayloadFrequency) {
+  MOZ_ASSERT(NS_IsMainThread());
+  mPayloadType = Some(aPayloadType);
+  mPayloadFrequency = Some(aPayloadFrequency);
+}
+
 void RTCDTMFSender::InsertDTMF(const nsAString& aTones, uint32_t aDuration,
                                uint32_t aInterToneGap, ErrorResult& aRv) {
   if (!mTransceiver->CanSendDTMF()) {
@@ -129,9 +133,8 @@
     } else {
       // Reset delay if necessary
       StartPlayout(mDuration + mInterToneGap);
-      // Note: We default to channel 0, not inband, and 6dB attenuation.
-      //      here. We might want to revisit these choices in the future.
-      mConduit->InsertDTMFTone(0, tone, true, mDuration, 6);
+      mDtmfEvent.Notify(DtmfEvent(mPayloadType.ref(), mPayloadFrequency.ref(),
+                                  tone, mDuration));
     }
   }
 